atlite/data.py

```diff
@@ -124,28 +128,28 @@ def cutout_prepare(cutout, features=slice(None), tmpdir=None, overwrite=False):
         logger.info(f'Calculating and writing with module {module}:')
         missing_features = missing_vars.index.unique('feature')
         ds = get_features(cutout, module, missing_features, tmpdir=tmpdir)
-        # make sure we don't loose any unused dimension by selecting
-        ds = ds[missing_vars.values].assign_coords(ds.coords)
-
-        ds = ds.assign_attrs(**cutout.data.attrs)
         prepared |= set(missing_features)
-        ds = ds.assign_attrs(prepared_features = list(prepared))
-        # convert bool to int for netCDF4 storing
-        ds.attrs.update({k: v if not isinstance(v, bool) else int(v)
-                         for k,v in ds.attrs.items()})
 
+        ds = cutout.data.merge(ds[missing_vars.values])\
+                            .assign_attrs(**cutout.data.attrs, **ds.attrs)
+        ds.attrs = non_bool_dict(cutout.data.attrs)
+        ds.attrs.update({'prepared_features': list(prepared)})
+
+        # write data to tmp file, copy it to original data, this is much saver
+        # than appending variables
+        tstr = next(tmpname())
+        tmp = cutout.path.parent / f'tmp_{tstr}_{cutout.path.stem}.nc'
         with ProgressBar():
-            if cutout.path.exists():
-                mode = 'a'
-            else:
-                mode = 'w'
+            ds.to_netcdf(tmp)
+            ds.close()
+        cutout.data.close()
+        if cutout.path.exists():
+            cutout.path.unlink()
+        tmp.rename(cutout.path)
 
-            ds.to_netcdf(cutout.path, mode=mode)
+        cutout.data = xr.open_dataset(cutout.path, chunks=cutout.chunks)
 
     if not keep_tmpdir:
         rmtree(tmpdir)
```

Comment:
`tmpdir` cleanup should be in a `finally`-close or an exception will leak the tmpdir.